.github/workflows/check_license_header.yaml

```diff
@@ -0,0 +1,85 @@
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+                   -o  -name 'CMakeLists.txt'                                     \
+               ')'                                                                \
+            > result-extension.check
+          # Generate an error message for each file with an unkown extension
+          for filepath in `cat result-extension.check`; do                        \
+            echo "::error file=$filepath,line=1::check extension of $filepath";   \
+          done
+
+          test ! -s result-extension.check
+
+      - name: Check source code
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Compare first lines of each source file with reference in misc/HEADER
+          find src include test miniapp -type f                                   \
+               '('                                                                \
+                       -name '*.cpp'                                              \
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+               ')'                                                                \
+            | xargs -I{} sh -c                                                    \
+                ".github/check_license.sh misc/HEADER {} > /dev/null || echo {}"  \
+            > result-cpp.check
+
+          # Generate an error message for each offending file
+          for filepath in `cat result-cpp.check`; do                              \
+            echo "::error file=$filepath,line=1::check the license in $filepath"; \
+          done
+
+          test ! -s result-cpp.check
+
+      - name: Check CMake and Python files
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Generate header license CMake/Python compliant (with hashes)
+          sed 's|^//|#|g' misc/HEADER > HEADER_HASH
+
+          # Check license in CMake files
+          find  '(' -type f ! -path './misc*' ')'                                 \
+                '('                                                               \
+                        -name 'CMakeLists.txt'                                    \
+                    -o  -name 'DLAF*.cmake'                                       \
+                    -o  -name 'DLAF*.cmake.in'                                    \
```

Comment:
I don't see any special reason for FindCMake, I just skipped them previously when we didn't have the special tags yet.

About others, just for reference a list grouped by file extensions.

```
# *.sh
./ci/ctest_to_gitlab.sh
./ci/set_github_status.sh
./scripts/build.sh
./.github/check_license.sh
./.github/format.sh
```

```
# *.yaml
./ci/docker/gpu-debug.yaml
./ci/docker/gpu-release.yaml
./ci/docker/cpu-debug.yaml
./ci/docker/cpu-release.yaml
./.github/workflows/check_license_header.yaml
./spack/repo.yaml
```
For both shell and yaml, I don't see any obstacle.

```
# *.py
./scripts/miniapps.py
./scripts/plot_hegst_weak.py
./scripts/plot_hegst_strong.py
./scripts/systems.py
./scripts/plot_chol_weak.py
./scripts/plot_trsm_strong.py
./scripts/postprocess.py
./scripts/gen_weak.py
./scripts/plot_chol_strong.py
./scripts/plot_trsm_weak.py
./scripts/gen_strong.py
./spack/packages/dla-future/package.py
```

About python scripts, we already have the check for the ones inside `scripts` folder. The other one is the spack package, which has the spack license, but I don't know if we are allowed to add ETHZ one. Anyone has any info on this side? However for the moment I think that since it is not already public in the official spack repo, we can do whatever we want with this file (adding a special tag, using just our license, ...).

TL;DR We can add the check for all of them (they all use hash for comments, so there are no big changes to do) and eventually skip it for specific files.

Reply:
If it's straightforward enough to add I think it makes sense to check all of them. No knowledge about the spack package but I'd imagine it should possibly have both. Easiest might be to keep the LLNL one and skip the check.

Reply:
- Error: check the license in ./cmake/FindLAPACK.cmake
it comes from CMake Recipes (ETHZ), we can just add `dlaf-no-check-license`
- Error: check the license in ./cmake/FindMKL.cmake
it comes from CMake Recipes (ETHZ), we can just add `dlaf-no-check-license`

About next twos, as already said, it is a transitional setup, until we will release it in the official spack repo, when we can drop the `repo.yaml`, and `dla-future/package.py` will probably require to have just a single spack-related license.

Until then, we can decide to have 
- Error: check the license in ./spack/repo.yaml
We can simply add our license, even if it is a minor file.
- Error: check the license in ./spack/packages/dla-future/package.py
We can have it multi-license for the moment.

Waiting for your feedback @msimberg @rasolca @teonnik before these last fixes.

Reply:
I don't think this is a blocker but you could have a "minimal license check" which only checks that the file contains `SPDX-License-Identifier:` and would at least somewhat cover the corner cases. But I think what you've suggested in the above comment is also reasonable.